File: src/python/parser.py
import time
import os

start_time = time.time()


def main(index):
    print(f'attempt {index}')
    source_file_path = f'{os.getcwd()}/data/data-source.csv'
    target_file_path = f'{os.getcwd()}/data/data-target.csv'
    source_file = open(source_file_path, "r")
    target_file = open(target_file_path, "w")
    target_file.truncate(0)

    for line in source_file.readlines():
        target_file.write(line)

    # Release used resources
    source_file.close()
    target_file.close()
    print(time.time() - start_time)


if __name__ == '__main__':
    main(10)

File: src/go/parser.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"time"
)

func main() {
	start := time.Now()
	sourceDir := "/Users/AndreyFurman/Downloads/parser-metrics/data/"
	sourceFilePath := fmt.Sprintf("%sdata-source.csv", sourceDir)
	targetFilePath := fmt.Sprintf("%sdata-target.csv", sourceDir)
	sourceFile, errSourceFile := os.Open(sourceFilePath)

	if errSourceFile != nil {
		log.Fatal(fmt.Sprintf("Error on open source file: %s", errSourceFile))
	}
	defer func(sourceFile *os.File) {
		err := sourceFile.Close()
		if err != nil {
			log.Fatal(fmt.Sprintf("Error on close source file: %s", err))
		}
	}(sourceFile)
	targetFile, errTargetFile := os.OpenFile(targetFilePath, os.O_TRUNC|os.O_APPEND|os.O_WRONLY, os.ModeAppend)
	if errTargetFile != nil {
		log.Fatal(fmt.Sprintf("Error on open target file: %s", errTargetFile))
	}
	scanner := bufio.NewScanner(sourceFile)
	for scanner.Scan() {
		_, err := targetFile.WriteString(fmt.Sprintf("%s\n", scanner.Text()))
		if err != nil {
			log.Fatal(fmt.Sprintf("Error on write file string %s", err))
		}
	}
	fmt.Println(time.Since(start))
}
